presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -1066,6 +1066,61 @@ protected Object visitLikePredicate(LikePredicate node, Object context)
                     optimizedEscape);
         }
 
+        private boolean isLikePattern(Slice pattern, Slice escape)
+        {
+            String stringPattern = pattern.toStringUtf8();
+            if (escape == null) {
+                return stringPattern.contains("%") || stringPattern.contains("_");
+            }
+
+            String stringEscape = escape.toStringUtf8();
+            checkArgument(stringEscape.length() == 1, "Escape must be empty or a single character");
+
+            char escapeChar = stringEscape.charAt(0);
+            boolean escaped = false;
+            boolean isLikePattern = false;
+            for (int currentChar : stringPattern.codePoints().toArray()) {
+                if (!escaped && (currentChar == escapeChar)) {
+                    escaped = true;
+                }
+                else if (escaped) {
+                    checkArgument(currentChar == '%' || currentChar == '_' || currentChar == escapeChar, "Escape character must be followed by '%', '_' or the escape character itself");
+                    escaped = false;
+                }
+                else if ((currentChar == '%') || (currentChar == '_')) {
+                    isLikePattern = true;
+                }
+            }
+            checkArgument(!escaped, "Escape character must be followed by '%', '_' or the escape character itself");
+            return isLikePattern;
+        }
+
+        private Slice unescapeLikePattern(Slice pattern, Slice escape)
+        {
+            if (escape == null) {
+                return pattern;
+            }
+
+            String stringEscape = escape.toStringUtf8();
+            checkArgument(stringEscape.length() == 1, "Escape must be empty or a single character");
```

Comment:
This can be removed since this has already been validated (it's confusing to have this validation but not others)